engine/src/main/java/org/terasology/engine/modes/StateMainMenu.java

```diff
@@ -121,6 +130,26 @@ public void init(GameEngine gameEngine) {
         if (!messageOnLoad.isEmpty()) {
             nuiManager.pushScreen(MessagePopup.ASSET_URI, MessagePopup.class).setMessage("Error", messageOnLoad);
         }
+
+        config = context.get(Config.class);
+        TelemetryConfig telemetryConfig = config.getTelemetryConfig();
+        LaunchPopupConfig launchPopupConfig = config.getLaunchPopupConfig();
+        if (!launchPopupConfig.isDisableLaunchPopup()) {
+            String telemetryTitle = "Telemetry In Terasology";
+            String telemetryMessage = "Telemetry system will send metrics and errors to the server. We do our best to avoid sending anything identifiable from your PC, e.g. telemetry includes your OS name, Java version but it doesn't include the Mac address, the home path, etc. You can review in Metric Menu and let us know if we missed something. Telemetry is super useful to an all volunteer project like ours and we'd really appreciate it!";
+            LaunchPopup telemetryConfirmPopup = nuiManager.pushScreen(LaunchPopup.ASSET_URI, LaunchPopup.class);
+            telemetryConfirmPopup.setMessage(telemetryTitle, telemetryMessage);
+            telemetryConfirmPopup.setYesHandler(() -> {
+                telemetryConfig.setEnableTelemetry(true);
+            });
+            telemetryConfirmPopup.setNoHandler(() -> {
+                telemetryConfig.setEnableTelemetry(false);
+            });
+            telemetryConfirmPopup.setOptionButtonText("Metric Menu");
+            telemetryConfirmPopup.setOptionHandler(()-> {
+                nuiManager.pushScreen(TelemetryScreen.ASSET_URI,TelemetryScreen.class);
+            });
+        }
```

Comment:
Everytime i see a method with more than 10-20 lines I feel like i have to do something.
This monster makes me feel physical pain 😧 

-> at least extract a method here 😄 or as uncle bob said: "Leave the campground cleaner than the way you found it"